src/components/filterdialog/filterdialog.js

```diff
@@ -115,6 +126,13 @@ function updateFilterControls(context, options) {
 function triggerChange(instance) {
     stopMultiSelect();
     Events.trigger(instance, 'filterchange');
+    // show or hide reset filter button on any filter change
+    const elem = document.querySelector('.btnResetAllFilters');
+    if (getFilterStatus(instance.options.query) === true) {
+        elem.classList.remove('hide');
+        return;
+    }
+    elem.classList.add('hide');
```

Comment:
Why not call `renderResetButton` here?

You should also narrow the scope - `document` may contain `btnResetAllFilters` from another page.

Reply:
@dmitrylyzo thanks four your valuable feedback!  
done in cd152f20334107bf31eb61cadf991789757142a2